bug 1525719: reorganize doc API data

escattone opened the pull request:
This PR re-organizes the document API JSON structure according to https://github.com/mozilla/kuma/pull/5371#discussion_r280120646. @davidflanagan, I think I've updated all of the appropriate places in the React files. I noticed that `docker-compose exec web npm run test` doesn't pass, but it doesn't seem to be related to my changes since the same failures occur when I run it with `master`.

davidflanagan commented:
It looks like  kuma/javascript/src/flow-typed/npm/jest_v24.x.x.js got reformated by prettier, and the indentation changed. You'll want to check out the original version of that file, and modify the definition of the format script in package.json so that it excludes the flow-typed directory, I suppose.

davidflanagan commented on kuma/javascript/src/page.jsx:
```diff
@@ -245,7 +245,9 @@ function Article() {
-                    dangerouslySetInnerHTML={{ __html: documentData.bodyHTML }}
+                    dangerouslySetInnerHTML={{
+                        __html: documentData.bodyHTML
+                    }}
```
I think you might need to re-run `npm run format` to put this line back the way it was. Since the text has not changed, I think the formatting should not have either.

davidflanagan commented on kuma/javascript/src/document-provider.jsx:
```diff
@@ -87,12 +86,15 @@ export default function DocumentProvider(
+                        const {
+                            locale,
+                            slug,
+                            absoluteURL: receivedURL
```
Oh, I see. You're defining the constant receivedURL to the value of the absoluteURL property.  I never use advanced destructuring like this, because I never understand it!

So the code is fine. Stylistically, I'd recommend just saying `const receivedURL = documentData.absoluteURL` however.

escattone replied:
I just learned it, and thought it was interesting, but you're absolutely right. It's not clear, and I'm happy to change it, thanks for the suggestion!

escattone commented:
@davidflanagan Thanks for your review and our conversation earlier today! I think I've addressed everything, except for the reorganization of where the `requestLocale` should best live. I thought you'd be best to address that in a separate PR if you wish to do so. I've squashed all of the commits into one, and re-based on `master`.
